src/librc/librc-depend.c

```diff
@@ -22,7 +22,17 @@
 
 #define GENDEP          RC_LIBEXECDIR "/sh/gendepends.sh"
 
-#define RC_DEPCONFIG    RC_SVCDIR "/depconfig"
+#define RC_DEPCONFIG    "/depconfig"
+
+static char *configpath = NULL;
+static char*
+depconfigpath_get() {
```

Comment:
I'm not a fan of "global + get function". And global variables, especially in libraries, are just hazards waiting to happen. I'd much rather just let the caller provide the buffer + size, and make the function fill it in as needed.

Also [don't use empty argument list](https://nrk.neocities.org/articles/c-stop-writing-old-function-decel.html) for declaring a function that takes no argument. Explicitly use `void` (e.g `f(void)`) because it has different meaning in C.

Reply:
Replaced all the globals with statics, the only global that it's still there is a "static bool is_userprefix", as an attempt to generalize this patch set for more than just user services, but to work with any "prefix" file structure.